nsv13/code/modules/overmap/weapons/mines.dm

```diff
@@ -0,0 +1,99 @@
+/obj/structure/space_mine
+	name = "space mine"
+	desc = "Like a naval mine, but in space!"
+	icon = 'nsv13/icons/overmap/effects.dmi'
+	icon_state = "mine_syndicate"
+	anchored = TRUE
+	density = FALSE
+	layer = ABOVE_MOB_LAYER
+	animate_movement = NO_STEPS
+	max_integrity = 300
+	integrity_failure = 100
+	var/datum/star_system/current_system
+	var/faction = "syndicate" //evil mines
+	var/damage = 100
+	var/damage_type = BRUTE
+	var/damage_flag = "overmap_heavy"
+	alpha = 50 //They're supposed to be sneaky, their main advantage is being cloaked
+
+/obj/structure/space_mine/Initialize(mapload, var/list/coordinates, var/new_faction, var/datum/star_system/system)
+	. = ..()
+	if(system)
+		current_system = system
+	else //Someone is probably spawning us on the overmap, so we assume it's next to the main ship
```

Comment:
Feels unsafe to assume. Should probably either be systemless or attempt to find the system it belongs to.

Reply:
I haven't found a satisfying way to get the system of something not currently linked to anything else